### PR TITLE
SDD-837 - changed delete venue page

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Added tests for deleting an unpublished venue
- Logic to delete an unpublished venue
- Delete page.

### CODE REVIEW

**Review by soharadesign — APPROVED** [review]

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs [hunk 1]
```diff
@@ -185,6 +187,32 @@ private void CheckStreamIsProcessable(Stream stream)
             }
         }
 
+        public async Task<bool> DeleteVenueUploadRow(Guid venueUploadId, int rowNumber)
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs:190]: Would be good to make this `DeleteVenueUploadRowForProvider(Guid providerId, int rowNumber)` to align with other methods.

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs [hunk 2]
```diff
@@ -185,6 +187,32 @@ private void CheckStreamIsProcessable(Stream stream)
...
+        {
+            using (var dispatcher = _sqlQueryDispatcherFactory.CreateDispatcher())
+            {
+                var (existingRows, lastRowNumber) = await dispatcher.ExecuteQuery(new GetVenueUploadRows() { VenueUploadId = venueUploadId });
+                var venueUpload = await dispatcher.ExecuteQuery(new GetVenueUpload() { VenueUploadId = venueUploadId });
+                var rowToDelete = existingRows.FirstOrDefault(x => x.RowNumber == rowNumber);
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs:196]: nit
```suggestion
                var rowToDelete = existingRows.SingleOrDefault(x => x.RowNumber == rowNumber);
```

File: src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs [hunk 3]
```diff
@@ -21,5 +21,6 @@ public interface IFileUploadProcessor
         Task<UploadStatus> WaitForVenueProcessingToCompleteForProvider(Guid providerId, CancellationToken cancellationToken);
 
         Task<SaveFileResult> SaveCourseFileForProvider(Guid providerId, Stream stream, UserInfo uploadedBy);
+        Task<bool> DeleteVenueUploadRow(Guid venueUploadId, int rowNumber);
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs:24]: Nit: move into the section with other venue bits

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/Delete.cshtml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+﻿@model Dfc.CourseDirectory.WebV2.Features.DataManagement.Venues.Delete.Response
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/Delete.cshtml:1]: Could we call this view `DeleteRow.cshtml` so it's clearly differentiated between `DeleteUpload.cshtml`?

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/DeleteVenue.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+using Dfc.CourseDirectory.Core.DataStore.Sql;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Models;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using Dfc.CourseDirectory.Core.Validation;
+using FluentValidation.Results;
+using MediatR;
+using OneOf;
+using OneOf.Types;
+
+namespace Dfc.CourseDirectory.WebV2.Features.DataManagement.Venues.Delete
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/DeleteVenue.cs:16]: Similar to above, I think `DeleteRow` as a namespace (and file name) would be good here.

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/VenuesDataManagementController.cs [hunk 6]
```diff
@@ -203,5 +204,33 @@ public IActionResult Formatting()
         {
             return View();
         }
+
+        [HttpPost("resolve/delete")]
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/VenuesDataManagementController.cs:208]: Would be good to have this route as `resolve/{rowNumber}/delete` so it aligns with the error fix pages.

**Review by soharadesign — APPROVED** [review]

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs (hunk 1 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.cs]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs (hunk 3 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs]: done

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/Delete.cshtml (hunk 4 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/Delete.cshtml]: done

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/DeleteVenue.cs (hunk 5 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/DeleteVenue.cs]: done

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/VenuesDataManagementController.cs (hunk 6 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Venues/VenuesDataManagementController.cs]: done

**Pull request merged**